controller/src/main/java/com/esm/epam/controller/CertificateController.java

```diff
@@ -25,75 +26,98 @@
     @JsonView(View.UI.class)
-    public ResponseEntity<Certificate> getCertificate(@PathVariable("id") @Min(1L) Long id) throws ResourceNotFoundException, DaoException {
+    public ResponseEntity<Certificate> getCertificate(@PathVariable("id") @Min(1L) Long id) throws ResourceNotFoundException, DaoException, ControllerException, ServiceException {
         Certificate certificate = certificateService.getById(id);
+        hateoasBuilder.buildFullHateoas(certificate);
         return new ResponseEntity<>(certificate, HttpStatus.OK);
     }
 
 
     @DeleteMapping("/{id}/tags/{tag_id}")
-    public ResponseEntity<Certificate> deleteCertificate(@PathVariable("id") @Min(1L) Long id, @PathVariable("tag_id") @Min(1L) Long idTag) throws ResourceNotFoundException, DaoException {
-        return certificateService.deleteTag(id, idTag)
-                .map(ResponseEntity::ok)
-                .orElseGet(() -> ResponseEntity.noContent().build());
+    @JsonView(View.UI.class)
+    public ResponseEntity<RepresentationModel<Certificate>> deleteTagCertificate(@PathVariable("id") @Min(1L) Long id, @PathVariable("tag_id") @Min(1L) Long idTag) throws DaoException, ControllerException, ServiceException, ResourceNotFoundException {
+        ResponseEntity<RepresentationModel<Certificate>> responseEntity;
+        Optional<Certificate> addedCertificate = certificateService.deleteTag(id, idTag);
+        if (addedCertificate.isPresent()) {
+            hateoasBuilder.buildFullHateoas(addedCertificate.get());
+            responseEntity = new ResponseEntity<>(addedCertificate.get(), OK);
+        } else {
+            responseEntity = ResponseEntity.noContent().build();
+        }
+        return responseEntity;
     }
 
     @DeleteMapping("/{id}")
-    public ResponseEntity<Void> deleteCertificate(@PathVariable("id") @Min(1L) Long id) throws ResourceNotFoundException {
-        ResponseEntity<Void> responseEntity;
+    @JsonView(View.UI.class)
+    public ResponseEntity<RepresentationModel<Certificate>> deleteCertificate(@PathVariable("id") @Min(1L) Long id) throws ResourceNotFoundException, ControllerException, ServiceException, DaoException {
+        ResponseEntity<RepresentationModel<Certificate>> responseEntity;
         if (certificateService.deleteById(id)) {
-            responseEntity = new ResponseEntity<>(HttpStatus.OK);
+            RepresentationModel<Certificate> representationModel = new RepresentationModel<>();
+            hateoasBuilder.buildDefaultHateoas(representationModel);
+            responseEntity = new ResponseEntity<>(representationModel, OK);
         } else {
-            responseEntity = new ResponseEntity<>(HttpStatus.NO_CONTENT);
+            responseEntity = ResponseEntity.noContent().build();
         }
-
         return responseEntity;
     }
 
-
     @PostMapping
-    public ResponseEntity<Certificate> addCertificate(@Valid @RequestBody Certificate certificate, BindingResult bindingResult) throws DaoException {
```

Comment:
Why you are not using Data transfer objects (DTO). It's an important thing to split web and persistence layers. In your case, if the model changes it brakes web because automatically changed you response.